### PR TITLE
Use agpypeline

### PR DESCRIPTION
**Pull request by Chris-Schnaufer**
## Description

Changed template code to use AgPypeline library

## Types of changes

Moved configuration to derived class
Moved transformer code to derived class
Fixed up generate_dockerfile.py to use new configuration class
Minor update to README

## Checklist:

- [x] I have read the documentation on [CONTRIBUTING](https://github.com/AgPipeline/Organization-info) 
- [x] My updates require a change to the documentation 
- [x] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: configuration.py [hunk 1]
```diff
@@ -1,33 +1,38 @@
...
 
-# Short name of the transformer
-TRANSFORMER_NAME = ''
+    # The transformer description
+    transformer_description = ''
 
-# The sensor associated with the transformer
-TRANSFORMER_SENSOR = ''
+    # Short name of the transformer
+    TRANSFORMER_NAME = ''
```
Comment (jvanderleeuw) [inline configuration.py:16]: Should this be lowercase to correspond with the rest of the parameters?

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+agpypeline==0.0.20
```
Comment (jvanderleeuw) [inline requirements.txt:1]: This could potentially be changed to the latest version of agpypeline, 0.0.37

File: generate_docker.py [hunk 3]
```diff
@@ -55,8 +55,8 @@ def generate_dockerfile(base_image_name: str) -> None:
         with open(dockerfile_name, 'w') as out_file:
             for line in template:
                 if line.startswith('LABEL maintainer='):
-                    out_file.write("LABEL maintainer=\"{0} <{1}>\"\n".format(configuration.AUTHOR_NAME,
-                                                                             configuration.AUTHOR_EMAIL))
+                    out_file.write("LABEL maintainer=\"{0} <{1}>\"\n".format(ConfigurationInfo.author_name,
+                                                                             ConfigurationInfo.author_email))
                 elif line.startswith('FROM base-image'):
                     out_file.write("FROM {0}\n".format(base_image_name))
                 else:
```
Comment (jvanderleeuw) [inline generate_docker.py:62]: When the Dockerfile is generated, and then I try to build an image, there is a permissions error for transformer.py. I tried to solve this by adding in the permissions/changing lines from plot-base-rgb, but they did not solve the problem. Output is attached
Screen Shot 2020-07-23 at 9 00 31 AM
Screen Shot 2020-07-23 at 9 01 35 AM

File: configuration.py (hunk 1 above)
Comment (Chris-Schnaufer) [reply configuration.py:16]: Thanks for the catch

File: generate_docker.py (hunk 3 above)
Comment (Chris-Schnaufer) [reply generate_docker.py:62]: There were issue with the Dockerfile.template file that have now been resolved.

File: requirements.txt (hunk 2 above)
Comment (Chris-Schnaufer) [reply requirements.txt:1]: It hasn't been tested with that version. I will put in a ticket for updating the repos once we have a "final" agpypeline version.

**Pull request merged**